src/wallet/rpcdump.cpp

```diff
@@ -67,43 +67,47 @@ class GetSpendingKeyForPaymentAddress : public boost::static_visitor<libzcash::S
 
 enum SpendingKeyAddResult {
     KeyAlreadyExists,
-    Added,
-    NotAdded,
+    KeyAdded,
+    KeyNotAdded,
 };
 
 class AddSpendingKeyToWallet : public boost::static_visitor<SpendingKeyAddResult>
 {
 private:
     CWallet *m_wallet;
+    int64_t nTime;
+    bool log;
 public: 
-    AddSpendingKeyToWallet(CWallet *wallet) : m_wallet(wallet) {}
+    AddSpendingKeyToWallet(CWallet *wallet, int64_t _nTime, bool _log) : m_wallet(wallet), nTime(_nTime), log(_log) {}
```

Comment:
It is this way because `z_importwallet` logged each address before it imported it where as `z_importkey` didn't, and I wanted to be able to reuse the visitor we created for `z_importkey`.

I could change AddSpendingKeyToWallet to return a pair of the result and an optional address and remove the logging from inside the visitor which might be more correct, but feels clunkier. I'd also be happy to remove the logging, but I can see use cases where that might be helpful. Let me know what you think or if you have a better idea of how to do this.

Reply:
Ok, that makes sense.  Looks good then.